lib/utils/error.go

```diff
@@ -354,13 +355,22 @@ func ShouldReconnectPeer(err error) error {
 	return err
 }
 
+// NewFailedPreconditionError returns a new failed precondition error
+// with optional original error err
+func NewFailedPreconditionError(err error) error {
+	return WrapExitCodeError(defaults.FailedPreconditionExitCode, err)
+}
+
 // ExitCodeError defines an interface for exit code errors
 type ExitCodeError interface {
 	error
 	ExitCode() int
+	// OrigError returns the original error this error wraps.
+	// Implements trace.Wrappable
```

Comment:
Note: there's actually no `trace.Wrappable` but should be as implementing the complete `trace.Error` interface is overkill.

Reply:
Nit: the note compared to the comment would confuse me if I encountered the comment when coming across this later...